fms/utils/generation.py

```diff
@@ -51,13 +57,36 @@ def generate(
         num_beams: TODO: support beam search
         use_cache: requires that the model accept use_cache and
             past_key_value_states args in forward method.
+        pad_id: int
+            the pad token id to use in case of batch generation where sequences are of different lengths
```

Comment:
I'd add an extra optional parameter for using nested tensors if batched here, and for now default it to False until all the nested work gets in a stable PT release